### DISCUSSION TITLE
Vercel and maxDuration (also BUG in Vercel documentation)

### DISCUSSION THREAD

**Discussion by kyllerss** [sveltejs/kit#11946] [Q&A]
I have a single SvelteKit app deployed on Vercel. The app is deployed as a single function. Most of the endpoints are accessed by the the app client, so a short response duration (say 20 seconds) is what I need. However, I have other endpoints that are triggered by cron jobs that should run for 5 minutes and have different memory settings.

The documentation states that I can set the `maxDuration` configuration in the vercel adapter in `svelte.config.js`. How can I override the `maxDuration` for these cron jobs?

As a side note, I saw that the [documentation](https://vercel.com/docs/functions/configuring-functions/duration) states: 

> While Vercel specifies [defaults](https://vercel.com/docs/functions/runtimes#max-duration) for the maximum duration of a function, you can also override it in your vercel.json file with:
>
> This glob pattern will match everything in the specified path, so you may wish to be more specific by adding a file type, such as app/api/**/*.ts instead.  

The above is the resulting information when you select `SvelteKit` from the code sample dropdown selector - the code sample it references is missing for SvelteKit. Assuming I read in-between the lines, is the glob pattern that goes into the `vercel.json` file applicable to individual files even if the app is deployed as a single function? I haven't had much success changing the `maxDuration` value on anything other than the global setting in the adapter. 

Also, there is the SvelteKit `adapter-vercel` documentation that refers to overriding the configuration at a granular level using [exports](https://kit.svelte.dev/docs/adapter-vercel), but I have been unable to have those changes override the global `maxDuration` setting.

Any suggestions greatly appreciated!

**Comment by Rishab49** [comment]
Do you have vercel pro or enterprise subscription?

**Reply by kyllerss** [reply]
Yes, I do.

**Comment by kyllerss** [comment]
Any updates?

**Comment by jhoddevik** [comment]
I changed from @sveltejs/adapter-auto to @sveltejs/adapter-vercel, and then I got it working.

```
import adapter from '@sveltejs/adapter-vercel';
export default {
	kit: {
		adapter: adapter({
			maxDuration: 300,
		})
	}
};
```

**Reply by kyllerss** [reply]
Yup. I'm using the Vercel adapter too. I can specify global timeouts just like you did. The problem is that I can't override it at a function-level (which the documentation states one can do).

**Reply by jhoddevik** [reply]
Ah, yes. I gave up on the glob pattern. I was just happy to get the global value to work

**Comment by demanzonderjas** [comment]
From the [Svelte documentation](https://svelte.dev/docs/kit/adapter-vercel), the current solution is now to export a config object from your +server.js, +page(.server).js and +layout(.server).js (or .ts):

```
import type { Config } from '@sveltejs/adapter-vercel';

export const config: Config = {
	maxDuration: 100
};
```

**Reply by teemingc** [reply]
> Also, there is the SvelteKit adapter-vercel documentation that refers to overriding the configuration at a granular level using [exports](https://kit.svelte.dev/docs/adapter-vercel), but I have been unable to have those changes override the global maxDuration setting.

The route level config should override the adapter config.

https://github.com/sveltejs/kit/blob/9fbfb22e442236e1a8ffa27a71fbef84166646e6/packages/adapter-vercel/index.js#L234

https://svelte.dev/docs/kit/adapter-vercel#Deployment-configuration
